packages/eslint-plugin/src/util/getBaseTypesOfClassMember.ts

```diff
@@ -0,0 +1,42 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+import * as ts from 'typescript';
+
+/**
+ * Given a member of a class which extends another class or implements an interface,
+ * yields the corresponding member type for each of the base class/interfaces.
+ */
+export function* getBaseTypesOfClassMember(
+  checker: ts.TypeChecker,
+  memberTsNode: ts.MethodDeclaration | ts.PropertyDeclaration,
```

Comment:
Continuing from the comment on assertions: I'll also note that we generally try to work with the TSESTree AST rather than TypeScript's, when possible. We have more flexibility and power to make the AST more specific because we're the ones who define it (and we define it as a discriminated union).

In this case it'd be more typical for a typed lint rule to take in the TSESTree node. You can also pass in the `services: ParserServicesWithTypeInformation` to use APIs like `services.getSymbolAtLocation`.

Reply:
Done